patches/kaggle_gcp.py

```diff
@@ -310,6 +310,24 @@ def init_natural_language():
     monkeypatch_client(language.LanguageServiceAsyncClient, kernel_credentials)
     return language
 
+def init_ucaip():
+    from google.cloud import aiplatform
+    if not is_user_secrets_token_set():
+        return
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_cloudai():
+        return
+
+    from kaggle_secrets import GcpTarget
+    from kaggle_gcp import KaggleKernelCredentials
+    kaggle_kernel_credentials = KaggleKernelCredentials(target=GcpTarget.CLOUDAI)
+
+    # Patch the 2 ucaip clients: Datasets & CustomTrainingJob
+    monkeypatch_client(aiplatform.Datasets, kaggle_kernel_credentials)
```

Comment:
The other clients all have the `*Client` suffix: `AutoMlClient`, `LanguageServiceClient`, `PredictionServiceClient`.

Did you check that `Datasets` and `CustomTrainingJob` uses the same structure and ensured that `monkeypatch_client` works for these classes?